### ISSUE TITLE
Spawner-less purchase bug

### ISSUE THREAD

**Issue by PikaMug** [ScreamingSandals/BedWars#386]
#### Describe the bug
Exception when attempting to purchase an item from Villager_shop when no Iron or Gold spawners have been set.

#### Server and plugin version
Spigot version: This server is running Paper version git-Paper-132 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: efd7e51)
BedWars version: 0.2.20

#### Installed plugins on server (can be obtained by `/pl`)

BedWars, Citizens, Essentials, EssentialsSpawn, GPS, GPS-Quests, Interactions, ItemsXL, LuckPerms, mcMMO, MobArena, Multiverse-Core, MythicMobs, Parties, Quests, Unite, Vault, WorldEdit, WorldGuard

#### Console log

```
[03:31:19 ERROR]: Could not pass event ShopTransactionEvent to BedWars v0.2.20
java.lang.NullPointerException: Cannot invoke "org.screamingsandals.bedwars.api.game.ItemSpawnerType.getStack(int)" because "type" is null
        at org.screamingsandals.bedwars.inventories.ShopInventory.handleBuy(ShopInventory.java:416) ~[?:?]
        at org.screamingsandals.bedwars.inventories.ShopInventory.onShopTransaction(ShopInventory.java:278) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor489.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.17.1.jar:git-Paper-132]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.17.1.jar:git-Paper-132]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.1.jar:git-Paper-132]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.17.1.jar:git-Paper-132]
        at org.screamingsandals.bedwars.lib.sgui.listeners.InventoryListener.onInventoryClick(InventoryListener.java:172) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor435.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.17.1.jar:git-Paper-132]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.17.1.jar:git-Paper-132]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.1.jar:git-Paper-132]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:2715) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:55) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:11) ~[?:?]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:36) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1348) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.server.MinecraftServer.shouldRun(MinecraftServer.java:190) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1327) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1320) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[?:?]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1281) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1192) ~[patched_1.17.1.jar:git-Paper-132]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[patched_1.17.1.jar:git-Paper-132]
        at java.lang.Thread.run(Thread.java:831) [?:?]
```

#### Configuration (of config.yml and shop.yml)
config https://pastebin.com/AcDeA6zw
shop https://pastebin.com/CK58a8Lv

**Comment by zlataovce** [comment]
remove iron and gold references from your shop or create the spawners

**Comment by PikaMug** [comment]
Yes, I'm aware of the fix. I'm reporting this because there still ought to be a null check somewhere. But thank you!

**Comment by zlataovce** [comment]
a fix was implemented in a 0.2.21 snapshot
